Data/NF/Unsafe.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 708
+{-# LANGUAGE PatternSynonyms #-}
+#endif
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Data.NF.Unsafe
+-- Copyright   :  (c) David Feuer 2016
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  ezyang@cs.stanford.edu
+-- Stability   :  experimental
+-- Portability :  portable
+--
+-- This module provides an unsafe pattern synonym for the type 'NF', allowing you
+-- to create values of type 'NF' without any runtime overhead if you fulfill a
+-- proof obligation that the value is already in normal form. It also provides
+-- functions allowing more general coercions relating to 'NF'.
```

Comment:
Agreed; I'll change the names.

On Jul 22, 2016 6:18 PM, "Edward Z. Yang" <notifications@github.com> wrote:

> In Data/NF/Unsafe.hs
> <https://github.com/ezyang/nf/pull/6#discussion_r71954007>:
>
> > +#endif
> > +
> > +-----------------------------------------------------------------------------
> > +-- |
> > +-- Module      :  Data.NF.Unsafe
> > +-- Copyright   :  (c) David Feuer 2016
> > +-- License     :  BSD-style (see the file LICENSE)
> > +--
> > +-- Maintainer  :  ezyang@cs.stanford.edu
> > +-- Stability   :  experimental
> > +-- Portability :  portable
> > +--
> > +-- This module provides an unsafe pattern synonym for the type 'NF', allowing you
> > +-- to create values of type 'NF' without any runtime overhead if you fulfill a
> > +-- proof obligation that the value is already in normal form. It also provides
> > +-- functions allowing more general coercions relating to 'NF'.
>
> Then the names need to be disambiguated better. mappingCoercion is very
> unclear on its own.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ezyang/nf/pull/6/files/db57fa061540b46fb54928e8131e84be376b7fe9#r71954007>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABzi_XHAazNBFr1rZsG56jU8ti122v44ks5qYUGjgaJpZM4JTKLc>
> .
>